### ISSUE TITLE
Neo4j Spark Connector Problem 

### ISSUE THREAD

**Issue by nikoskous** [neo4j-contrib/neo4j-spark-connector#129]
Hello!
Im using the connector through Java programming language and versions:

_-Neo4j --> 3.5.8
-neo4j-spark-connector --> 2.4.1-M1
-Spark --> 2.4.0_

Im initialising the spark session like below: 

```
SparkSession spark = SparkSession
                .builder()
                .appName("Application Name").master("local")
                .config("spark.neo4j.bolt.user","neo4j")
                .config("spark.neo4j.bolt.password", "nikos")
                .config("spark.neo4j.bolt.url","bolt://localhost:7687")
                .getOrCreate();
```

and Im running those commands to get the results as DataFrame:

```
Neo4j neo = new Neo4j(getSparkSession().sparkContext());

Dataset<Row> o = neo.cypher("MATCH (n:Ship) WHERE n.CRAFT_ID <= {maxId} RETURN n.CRAFT_ID", scala.Predef$.MODULE$.Map().empty()).loadDataFrame();
```

But it throughs the following error:

```
20/06/07 13:39:03 ERROR Executor: Exception in task 0.0 in stage 0.0 (TID 0)
java.lang.NoSuchFieldError: NULL
	at org.neo4j.driver.v1.Values.<clinit>(Values.java:85)
	at org.neo4j.driver.v1.AuthTokens.basic(AuthTokens.java:72)
	at org.neo4j.driver.v1.AuthTokens.basic(AuthTokens.java:54)
	at org.neo4j.spark.Neo4jConfig.driver(Neo4jConfig.scala:15)
	at org.neo4j.spark.Neo4jConfig.driver(Neo4jConfig.scala:19)
	at org.neo4j.spark.Executor$.execute(Executor.scala:70)
	at org.neo4j.spark.rdd.Neo4jRDD.compute(Neo4jRDD.scala:17)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:121)
	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:402)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:408)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by conker84** [comment]
@nikoskous you need to specifi the `{maxId}` param

**Comment by nikoskous** [comment]
@conker84 When the {maxId} has type long it says that want object instead of long

`Required type: scala.collection.immutable.Map <String, Object>
Provided: java.util.Map <String, Long>
no instance(s) of type variable(s) K, V exist so that Map<K, V> conforms to Map<String, Object>`

When i cast the {maxId} to Object is throws this one: 

`Required type: scala.collection.immutable.Map <String, Object>
Provided: java.util.Map <String, Object>
no instance(s) of type variable(s) K, V exist so that Map<K, V> conforms to Map<String, Object`

**Comment by conker84** [comment]
@nikoskous since you're using JAVA your code should be changed into something like this:

```java
Neo4JavaSparkContext neo = Neo4JavaSparkContext.neo4jContext(getSparkSession().sparkContext());
// your maxId
long maxId = 10000L;
Dataset<Row> o = neo.cypher("MATCH (n:Ship) WHERE n.CRAFT_ID <= {maxId} RETURN n.CRAFT_ID", Collections.singletonMap("maxId", maxId)).loadDataFrame();
```

**Comment by nikoskous** [comment]
@conker84 I found the problem. 
I am using the neo4j-java-driver version 4.0.0, which is not compatible with the neo4j-spark-connector latest version, according to maven repo.
Are you planning to make an update to support the newest version of neo4j-java-driver?

**Comment by conker84** [comment]
@nikoskous we just released the a new version that supports Neo4j 4.0:
https://github.com/neo4j-contrib/neo4j-spark-connector/releases/tag/2.4.5-M1
But please, consider my previous suggests because you should write the job as I said.

**Comment by nikoskous** [comment]
@conker84 Thank you very much! Everything worked perfect

> Neo4JavaSparkContext neo = Neo4JavaSparkContext.neo4jContext(getSparkSession().sparkContext());
        long maxId = 50L;
        Dataset<Row> o = neo.queryDF("MATCH (n:Ship) WHERE n.CRAFT_ID > $maxId RETURN n.CRAFT_ID", Collections.singletonMap("maxId", maxId)).toDF("max");
        o.printSchema();
        o.show();
